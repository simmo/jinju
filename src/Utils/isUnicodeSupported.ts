import { isCI } from './isCI.js';

export const isUnicodeSupported = () => {
	if (process.platform !== 'win32') return process.env.TERM !== 'linux';

	return isCI || process.env.TERM_PROGRAM === 'vscode';
};
