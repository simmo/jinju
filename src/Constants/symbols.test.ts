import isUnicodeSupported from 'is-unicode-supported';
import { beforeEach, describe, expect, it, vi } from 'vitest';

vi.mock('is-unicode-supported');

describe('symbols', () => {
	beforeEach(() => {
		vi.resetModules();
	});

	it('returns an object of standard symbols', async () => {
		vi.mocked(isUnicodeSupported).mockReturnValue(true);

		const { Symbols } = await import('./symbols');

		expect(Symbols).toMatchSnapshot();
	});

	it('returns an object of fallback symbols', async () => {
		vi.mocked(isUnicodeSupported).mockReturnValue(false);

		const { Symbols } = await import('./symbols');

		expect(Symbols).toMatchSnapshot();
	});
});
