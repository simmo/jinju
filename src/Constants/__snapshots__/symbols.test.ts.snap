// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`symbols > returns an object of fallback symbols 1`] = `
{
  "Celebrate": "*",
  "Debug": "•",
  "Error": "×",
  "Info": "i",
  "Question": "?",
  "Success": "√",
  "Warning": "‼",
}
`;

exports[`symbols > returns an object of standard symbols 1`] = `
{
  "Celebrate": "★",
  "Debug": "•",
  "Error": "✖",
  "Info": "ℹ",
  "Question": "◆",
  "Success": "✔",
  "Warning": "⚠",
}
`;
